Sources/Features/Network/MultipartRequest.swift

```diff
@@ -48,41 +48,85 @@ extension MultipartRequest {
         var multipartRequest = urlRequest
         let boundary = generateBoundaryString()
         multipartRequest.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
-
-        let boundaryPrefix = "--\(boundary)\r\n"
-        let body = NSMutableData()
-        // attributes
-        for attribute in report.attributes {
-            body.appendString(boundaryPrefix)
-            body.appendString("Content-Disposition: form-data; name=\"\(attribute.key)\"\r\n\r\n")
-            body.appendString("\(attribute.value)\r\n")
-        }
-        // report file
-        body.appendString(boundaryPrefix)
-        body.appendString("Content-Disposition: form-data; name=\"upload_file\"; filename=\"upload_file\"\r\n")
-        body.appendString("Content-Type: application/octet-stream\r\n\r\n")
-        body.append(report.reportData)
-        body.appendString("\r\n")
-
-        // attachments
-        for attachment in Set(report.attachmentPaths).compactMap(Attachment.init(filePath:)) {
-            body.appendString(boundaryPrefix)
-            body.appendString("Content-Disposition: form-data; name=\"\(attachment.filename)\"; filename=\"\(attachment.filename)\"\r\n")
-            body.appendString("Content-Type: \(attachment.mimeType)\r\n\r\n")
-            body.append(attachment.data)
-            body.appendString("\r\n")
+        
+        // temporary file to stream data
+        let tempURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(UUID().uuidString)
+        
+        do {
+            let fileCreated = FileManager.default.createFile(atPath: tempURL.path, contents: nil, attributes: nil)
+            if !fileCreated {
+                throw HttpError.fileCreationFailed(tempURL)
```

Comment:
it won't be captured by the L:111 - which means we will return a generic error rather this this one?